rse-bash.Rmd

```diff
@@ -2160,342 +2209,284 @@ $ grep -n "it" haiku.txt
 10:Today it is not working
 ```
 
-Here, we can see that lines 5, 9, and 10 contain the letters "it".
+Another is `-i`,
+which does case-insensitive matching:
 
 We can combine options (i.e. flags) as we do with other Unix commands.
 For example, let's find the lines that contain the word "the". We can combine
 the option `-w` to find the lines that contain the word "the" and `-n` to number the lines that match:
 
 ```shell
-$ grep -n -w "the" haiku.txt
+$ grep -i to haiku.txt
 ```
 
 ```text
-2:Is not the true Tao, until
-6:and the presence of absence:
+You bring fresh toner.
+Today it is not working
 ```
 
-Now we want to use the option `-i` to make our search case-insensitive:
+We can combine options as with other commands:
 
 ```shell
-$ grep -n -w -i "the" haiku.txt
+$ grep -i -n haiku.txt
 ```
 
 ```text
-1:The Tao that is seen
-2:Is not the true Tao, until
-6:and the presence of absence:
+3:You bring fresh toner.
+10:Today it is not working
 ```
 
-Now, we want to use the option `-v` to invert our search, i.e., we want to output
-the lines that do not contain the word "the".
+We can also invert the match—i.e., print lines that *don't* match the pattern—using `-v`:
 
 ```shell
-$ grep -n -w -v "the" haiku.txt
+$ grep -i -n -v to haiku.txt
 ```
 
 ```text
 1:The Tao that is seen
-3:You bring fresh toner.
+2:Is not the true Tao, until
 4:
 5:With searching comes loss
+6:and the presence of absence:
 7:"My Thesis" not found.
 8:
 9:Yesterday it worked
-10:Today it is not working
 11:Software is like that.
 ```
 
-To search for text within multiple files in the same directory,
-wildcards can be used when specifying the file names.
-For example,
-`grep "the" *.txt` would search all `.txt` files in the current directory.
-To search all files within a directory and its subdirectories,
```

Comment:
I would add these 4 lines back in. `grep -r` is useful and often needed in my experience